### ISSUE TITLE
Importing an API revision corresponding to an existing Deployment environment fails to deploy

### ISSUE THREAD

**Issue by uvindra** [wso2/product-apim-tooling#731]
**Description:**
If a deployed API revision is exported and then imported into another environment that has the same gateway environment configured, the API fails to get deployed even though it is imported successfully.

Following error can be seen in APIM logs without any further stack trace

```
GlobalThrowableMapper  Error while importing API: SZ2PXQ5BMQXUZ6FYAPI version: 1.0.0 Error occurred while creating a new revision for the API: SZ2PXQ5BMQXUZ6FYAPI
```

**OS, DB, other environment details and versions:**    
Ubuntu 20.04

APIM 9.0.182.SNAPSHOT

**Steps to reproduce:**
1. Create a new API revision and deploy it.
2. Export the specific revision(deployment_env.yaml in the exported project will contain the default environment)
3. Import the API to another APIM instance(which will also contain the default environment out of the box)

**Test case**
TestExportImportApiSameGWEnv

**Comment by wasuradananjith** [comment]
I tried to reproduce this issue master. I followed the below steps.

![commands](https://user-images.githubusercontent.com/25246848/140459977-3bab303d-744d-4a46-b61c-16d2009a0e5f.png)

After importing the API to the second environment, there weren't any logs as ```GlobalThrowableMapper  Error while importing API: SZ2PXQ5BMQXUZ6FYAPI version: 1.0.0 Error occurred while creating a new revision for the API: SZ2PXQ5BMQXUZ6FYAPI```. (Refer to the below screenshot of the logs of the second server)

![not reproducing](https://user-images.githubusercontent.com/25246848/140460103-6f14e867-d70c-4a45-bca7-95df18d04b07.png)

Further, if we check the Publisher UI, the API has been deployed correctly.

![deployed](https://user-images.githubusercontent.com/25246848/140460145-d4452397-17c8-4367-9a15-80f7f2e3e0b1.png)

I ran the test named **TestExportImportApiSameGWEnv** individually (and together with other tests as well), and it is passing now.

![image](https://user-images.githubusercontent.com/25246848/140460405-fb4566d6-ac63-4c04-b0bd-87e5c456c932.png)

When considering the above facts, seems the issue has been resolved. Are we good to close this @uvindra ?
